### PR TITLE
Remove Javascript from template files

### PR DESCRIPTION
**Pull request by jacobpenny**
This is a PR to remove Javascript from template files because:
- Javascript in template files can't be easily linted, unit tested, or compiled
- Since it can't be compiled it prevents us from using module loaders like Webpack in the future
- The `getJSDependencies` method is misleading because there is often a JS dependency tucked away in a template file, and the order of dependency isn't immediately clear

It is a lot of changes but focused in scope:
- JS code in template files was moved into separate `index.js` files
- We used query string parameters and DOM `data-` attributes as alternatives to templating backend variables into JS code. 
- Because `getJSDependencies` includes the script tags in `<head>` we wrapped entry points in `$(document).ready()`
- Since we can now compile this code we ditched the `React.createFactory` syntax in favour of JSX.

The majority of the modules are complete, though a few are left for a subsequent PR due to potential conflicts with other outstanding PRs. Because it touches many files we'd like to get this reviewed and merged as soon as possible.

Modules:
- [x] ~~acknowledgements~~
- [ ] brainbrowser
- [x] ~~bvl_feedback~~
- [x] ~~candidate_list~~
- [x] ~~candidate_parameters~~
- [x] ~~configuration~~
- [x] ~~conflict_resolver~~
- [x] ~~create_timepoint~~
- [x] ~~dashboard~~
- [x] ~~data_integrity_flag~~
- [x] ~~data_release~~
- [x] ~~data_team_helper~~
- [x] ~~datadict~~
- [x] ~~dataquery~~
- [x] ~~dicom_archive~~
- [x] ~~document_repository~~
- [x] ~~examiner~~
- [ ] final_radiological_review
- [x] genomic_browser ( @xlecours )
- [x] ~~help_editor~~
- [ ] imaging_browser
- [x] imaging_uploader (Taking care of it in separate PR - Alex)
- [x] ~~instrument_builder~~
- [x] ~~instrument_list~~
- [ ] instrument_manager
- [x] ~~issue_tracker~~
- [x] ~~media~~
- [ ] mri_violations (Wait for #2255 #2200 #2195)
- [x] ~~new_profile~~
- [x] ~~next_stage~~
- [x] ~~reliability~~
- [x] ~~server_processes_manager~~
- [ ] statistics
- [x] ~~survey_accounts~~
- [x] ~~timepoint_flag~~
- [x] ~~timepoint_list~~
- [x] ~~training~~
- [x] ~~user_accounts~~

---

### CODE REVIEW

File: modules/candidate_parameters/php/NDB_Form_candidate_parameters.class.inc [hunk 1]
```diff
@@ -148,9 +148,11 @@ class NDB_Form_Candidate_Parameters extends NDB_Form
      */
     function getJSDependencies()
     {
-        $factory = NDB_Factory::singleton();
-        $baseURL = $factory->settings()->getBaseURL();
-        $deps    = parent::getJSDependencies();
+        $factory    = NDB_Factory::singleton();
+        $baseURL    = $factory->settings()->getBaseURL();
+        $deps       = parent::getJSDependencies();
+        $useConsent = $this->tpl_data['useConsent'];
```
Comment (driusan) [inline modules/candidate_parameters/php/NDB_Form_candidate_parameters.class.inc]: This should probably go into the LorisHelper global object config() instead of a URL?

See here: https://github.com/aces/Loris/blob/17.1-dev/smarty/templates/main.tpl#L14

File: modules/media/jsx/mediaIndex.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* global DynamicDataTable MediaUploadForm formatColumn */
+
+$(function() {
+  loris.hiddenHeaders = ['Cand ID', 'Session ID'];
```
Comment (driusan) [inline modules/media/jsx/mediaIndex.js]: There shouldn't be spaces in CandID and SessionID to be consistent with the rest of LORIS

File: modules/media/php/NDB_Menu_Filter_media.class.inc [hunk 3]
```diff
@@ -198,11 +196,14 @@ class NDB_Menu_Filter_Media extends NDB_Menu_Filter
         $factory = NDB_Factory::singleton();
         $baseURL = $factory->settings()->getBaseURL();
         $deps    = parent::getJSDependencies();
+
         return array_merge(
             $deps,
             [
-             $baseURL . "/media/js/columnFormatter.js",
+             $baseURL . "/media/js/columnFormatter.js?hasWritePermission="
```
Comment (driusan) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: the LorisHelper global has a permission function. Can this just use that?

File: modules/media/jsx/mediaIndex.js (hunk 2 above)
Comment (jacobpenny) [inline modules/media/jsx/mediaIndex.js]: Agreed, but these fields were just moved and I think changing them is outside the scope of this PR. It would require more changes in the media module. IMO it should be done in another PR.

File: modules/media/php/NDB_Menu_Filter_media.class.inc (hunk 3 above)
Comment (jacobpenny) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: Good call

**Comment by MounaSafiHarab** [comment]
smaller pull requests will be sent instead
